File: fixtures/scale/python/large/cipherlab/checks/check_tallyhash.py
from cipherlab.ciphers.tallyhash import tallyhash_digest
from cipherlab.specs.tallyhash_spec import tallyhash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_tallyhash():
    spec = tallyhash_spec()
    for text in corpus_hash():
        first = tallyhash_digest(text)
        second = tallyhash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"

File: fixtures/scale/ruby/large/lib/check_tallyhash.rb
require_relative 'tallyhash_digest'
require_relative 'tallyhash_digest_pair'
require_relative 'corpus_hash'

def check_tallyhash
  corpus_hash.each do |text|
    first = tallyhash_digest(text)
    second = tallyhash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end
